pkg/exec/exec.go

```diff
@@ -116,6 +116,12 @@ func CombinedOutputLines(cmd Cmd) (lines []string, err error) {
 	return lines, err
 }
 
+// InheritOutputWithBuf sets cmd's output to write to the current process's stdout and stderr, and also writes the combined output to a buffer.
+func InheritOutputWithBuf(cmd Cmd, buf *bytes.Buffer) {
```

Comment:
buffer seems oddly specific, we should make it io.Writer instead?

Reply:
Replied in the other comment. We cannot use `io.Writer` here since we need to capture the output as well as printed in the stdout and stderr. Or do you mean `io.MultiWriter`?

Reply:
not sure why we can't use it here? 
`io.Writer` is an interface both `io.MultiWriter` and `*bytes.Buffer` satisfy that interface.

Reply:
Fixed.